documentation/compatibility_with_puppet_agent.markdown

```diff
@@ -0,0 +1,106 @@
+# Puppet 3 Backwards Compatibility
+
+Puppet Server 2.1 introduces a new feature that allows Puppet 3 agents to
+operate with puppetserver.  Puppet Server 1.x requires Puppet 3 while Puppet
+Server 2.x requires Puppet 4.  These requirements created a situation where
+only Puppet 4 agents operate with Puppet Server 2.0 masters.  We've added
+Puppet 3 support in Puppet Server 2.1 to ease the upgrade process for users.
+
+Compatibility is provided by a so-called `legacy-routes-service`.  This service
+intercepts all Puppet 3 REST API requests and transforms the URL structure and
+request headers into a Puppet 4 compatible request which is handled normally by
+the Puppet Master service.  The biggest implication of this implementation is
+that Puppet's `auth.conf` rules need special attention if they've been
+customized from their default values because all Puppet 3 requests have already
+been transformed into Puppet 4 requests by the time `auth.conf` rules are
+enforced.
+
+Other implications of this implementation are discrepancies in the request and
+response REST API headers.   These differences are worth noting, but should
+have no effect on typical deployments.
+
+# Configuration settings
+
+When operating Puppet 3 agents against Puppet Server 2.1, some configuration
+changes on the agents may be required ahead of time.  Two settings in
+particular should be configured:
+
+ * `stringify_facts = false` which is required with a Puppet 4 master.
+ * The master will operate with the future parser turned on, so if the future
+   parser is not being used with Puppet 3, please turn it on and try it out
+   before upgrading to Puppet Server 2.1.  Any incompatibilities between
+   existing Puppet code and the future parser should be resolved prior to
+   upgrading to Puppet Server 2.1.
+
+# auth.conf
+
+The REST API URL structure has changed starting in Puppet 4.  The major change
+is the introduction of a versioned API rooted at `/puppet`.  In the general
```

Comment:
Could mention that CA-related routes are rooted at `/puppet-ca` whereas all other master routes would be rooted at `/puppet`.